plat/kvm/efi.c

```diff
@@ -349,18 +374,24 @@ static void uk_efi_setup_bootinfo_mrds(struct ukplat_bootinfo *bi)
 			continue;
 
 		rc = ukplat_memregion_list_insert(&bi->mrds,  &mrd);
-		if (unlikely(rc < 0))
-			uk_efi_crash("Failed to insert mrd\n");
+		if (unlikely(rc < 0)) {
+			uk_pr_err("Failed to insert mrd\n");
+			_uk_efi_crash();
```

Comment:
Why use `uk_pr_err` :thinking:? It is supposedly not initialized yet. Rely on just the proposed `UK_EFI_CRASH` instead.

Reply:
After exiting Boot Service, the EFI printing service does not work anymore. What do you think is a good way to handle this?

Reply:
Ah, right, this now gets executed after `ExitBootServices`... :man_facepalming:. Thanks for pointing it out :+1:.

Right... ok, well, our printing functionality on x86 is either the VGA Text Mode framebuffer which probably is not really there on UEFI (but its zone gets reserved anyway) or the good ol' "COM" UART which should always be there (I don't know of any chipset that does not initially hardcode it to those I/O ports, though might be different for some embedded x86 SoC's which we don't target anyway). On the other hand, on ARM, we actually do have something of an "early UART" and those console drivers **inefficiently** always check if the console is initialized before trying to print (for every character, ughhh...).

Therefore, I guess it's okay for now, after all, to leave it as just `uk_pr_err` + `do_uk_efi_crash` since nothing should break anyway.